backends/tfhe-cuda-backend/cuda/src/crypto/keyswitch.cuh

```diff
@@ -165,16 +183,36 @@ __global__ void keyswitch_gemm_copy_negated_message_with_indices(
   uint32_t lwe_in_idx = lwe_input_indices[lwe_id];
   uint32_t lwe_out_idx = lwe_output_indices[lwe_id];
 
+  Torus body_in =
+      lwe_in[lwe_in_idx * (lwe_dimension_in + 1) + lwe_dimension_in];
+  Torus body_out;
+  if constexpr (std::is_same_v<KSTorus, Torus>) {
+    body_out = -body_in;
+  } else {
+    body_out = closest_repr(
+        lwe_in[lwe_in_idx * (lwe_dimension_in + 1) + lwe_dimension_in],
+        sizeof(KSTorus) * 8, 1);
+
+    // Power of two are encoded in the MSBs of the types so we need to scale
+    // the type to the other one without having to worry about the moduli
+    Torus input_to_output_scaling_factor =
+        (sizeof(Torus) - sizeof(KSTorus)) * 8;
+
+    auto rounded_downscaled_body =
+        (KSTorus)(body_out >> input_to_output_scaling_factor);
+
+    body_out = -rounded_downscaled_body;
```

Comment:
Same here @IceTDrinker, maybe someone else from the CPU team knows better about KS32 but I would appreciate to have an external pair of eyes to confirm this is the correct logic.